make vet fmt is broken

```
$ make vet fmt
cd /Users/dims/go/src/git.openstack.org/openstack/openstack-cloud-controller-manager && go vet ./...
# git.openstack.org/openstack/openstack-cloud-controller-manager/pkg/volume/cinder/provisioner
pkg/volume/cinder/provisioner/iscsi_test.go:61:4: err declared but not used
pkg/volume/cinder/provisioner/iscsi_test.go:105:4: err declared but not used
pkg/volume/cinder/provisioner/iscsi_test.go:153:4: err declared but not used
pkg/volume/cinder/provisioner/rbd_test.go:78:4: err declared but not used
vet: typecheck failures
make: *** [vet] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -16,6 +16,7 @@ DEST := $(GOPATH)/src/$(GIT_HOST)/openstack/$(BASE_DIR)
 DEST := $(GOPATH)/src/$(PKG)
 SOURCES := $(shell find $(DEST) -name '*.go')
 HAS_GLIDE := $(shell command -v glide;)
+HAS_LINT := $(shell command -v golint;)
 
 GOOS ?= $(shell go env GOOS)
 VERSION ?= $(shell git describe --exact-match 2> /dev/null || \
@@ -81,6 +82,10 @@ fmt: work
 	cd $(DEST) && hack/verify-gofmt.sh
 
 lint: work
+ifndef HAS_LINT
+		go get -u github.com/golang/lint/golint
+		echo "installing lint"
+endif
 	cd $(DEST) && hack/verify-golint.sh
 
 vet: work

--- a/pkg/csi/cinder/controllerserver_test.go
+++ b/pkg/csi/cinder/controllerserver_test.go
@@ -19,8 +19,8 @@ package cinder
 import (
 	"testing"
 
-	"github.com/container-storage-interface/spec/lib/go/csi/v0"
 	"git.openstack.org/openstack/openstack-cloud-controller-manager/pkg/csi/cinder/openstack"
+	"github.com/container-storage-interface/spec/lib/go/csi/v0"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 )

--- a/pkg/csi/cinder/nodeserver_test.go
+++ b/pkg/csi/cinder/nodeserver_test.go
@@ -19,8 +19,8 @@ package cinder
 import (
 	"testing"
 
-	"github.com/container-storage-interface/spec/lib/go/csi/v0"
 	"git.openstack.org/openstack/openstack-cloud-controller-manager/pkg/csi/cinder/mount"
+	"github.com/container-storage-interface/spec/lib/go/csi/v0"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 )

--- a/pkg/identity/keystone/authenticator.go
+++ b/pkg/identity/keystone/authenticator.go
@@ -95,8 +95,8 @@ func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticateToken(token stri
 
 	extra := map[string][]string{
 		"alpha.kubernetes.io/identity/roles":        roles,
-		"alpha.kubernetes.io/identity/project/id":   []string{obj.Token.Project.Id},
-		"alpha.kubernetes.io/identity/project/name": []string{obj.Token.Project.Name},
+		"alpha.kubernetes.io/identity/project/id":   {obj.Token.Project.Id},
+		"alpha.kubernetes.io/identity/project/name": {obj.Token.Project.Name},
 	}
 
 	authenticated_user := &user.DefaultInfo{

--- a/pkg/volume/cinder/attacher.go
+++ b/pkg/volume/cinder/attacher.go
@@ -23,13 +23,13 @@ import (
 	"strings"
 	"time"
 
+	volumeutil "git.openstack.org/openstack/openstack-cloud-controller-manager/pkg/volume/util"
 	"github.com/golang/glog"
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/kubernetes/pkg/util/mount"
 	"k8s.io/kubernetes/pkg/volume"
-	volumeutil "git.openstack.org/openstack/openstack-cloud-controller-manager/pkg/volume/util"
 )
 
 type cinderDiskAttacher struct {

--- a/pkg/volume/cinder/cinder_util.go
+++ b/pkg/volume/cinder/cinder_util.go
@@ -24,9 +24,9 @@ import (
 	"strings"
 	"time"
 
+	volutil "git.openstack.org/openstack/openstack-cloud-controller-manager/pkg/volume/util"
 	"github.com/golang/glog"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	volutil "git.openstack.org/openstack/openstack-cloud-controller-manager/pkg/volume/util"
 
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/sets"

--- a/pkg/volume/cinder/provisioner/iscsi_test.go
+++ b/pkg/volume/cinder/provisioner/iscsi_test.go
@@ -82,6 +82,7 @@ var _ = Describe("Iscsi Mapper", func() {
 			Expect(source.ISCSI.IQN).To(Equal("iqn"))
 			Expect(source.ISCSI.Lun).To(Equal(int32(3)))
 			Expect(source.ISCSI.SessionCHAPAuth).To(BeFalse())
+			Expect(err).NotTo(HaveOccurred())
 		})
 
 		Context("when CHAP authentication is enabled", func() {
@@ -131,16 +132,19 @@ var _ = Describe("Iscsi Mapper", func() {
 				pass := string(cb.CreatedSecret.Data["node.session.auth.password"][:])
 				Expect(user).To(Equal("user"))
 				Expect(pass).To(Equal("pass"))
+				Expect(err).NotTo(HaveOccurred())
 			})
 
 			It("should target the namespace where the PVC resides", func() {
 				Expect(cb.Namespace).To(Equal(options.PVC.Namespace))
+				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 
 		Context("when the connection does not require CHAP authentication", func() {
 			It("should not create a CHAP secret", func() {
 				Expect(cb.CreatedSecret).To(BeNil())
+				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 	})
@@ -174,12 +178,14 @@ var _ = Describe("Iscsi Mapper", func() {
 			It("should delete the secret", func() {
 				Expect(cb.DeletedSecret).To(Equal("secretName"))
 				Expect(cb.Namespace).To(Equal("testNs"))
+				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 
 		Context("when the PV does not contain a secret reference", func() {
 			It("should not delete any secret", func() {
 				Expect(cb.DeletedSecret).To(BeEmpty())
+				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 	})

--- a/pkg/volume/cinder/provisioner/rbd_test.go
+++ b/pkg/volume/cinder/provisioner/rbd_test.go
@@ -102,6 +102,7 @@ var _ = Describe("Rbd Mapper", func() {
 				Expect(source.RBD.RBDImage).To(Equal("image"))
 				Expect(source.RBD.RadosUser).To(Equal("admin"))
 				Expect(source.RBD.SecretRef.Name).To(Equal("storageclass-cephx-secret"))
+				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 
@@ -112,6 +113,7 @@ var _ = Describe("Rbd Mapper", func() {
 
 			It("should be nil", func() {
 				Expect(source).To(BeNil())
+				Expect(err).To(HaveOccurred())
 			})
 		})
 
@@ -123,6 +125,7 @@ var _ = Describe("Rbd Mapper", func() {
 
 			It("should be nil", func() {
 				Expect(source).To(BeNil())
+				Expect(err).To(HaveOccurred())
 			})
 		})
 
@@ -133,6 +136,7 @@ var _ = Describe("Rbd Mapper", func() {
 
 			It("should have an image field containing a '/'", func() {
 				Expect(source.RBD.RBDImage).To(Equal("image/foo"))
+				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 	})

